src/java/org/apache/cassandra/db/guardrails/Values.java

```diff
@@ -31,35 +31,60 @@
 import static java.lang.String.format;
 
 /**
- * A guardrail that warns about but ignores some specific values, and rejects the use of some other values.
+ * A guardrail that warns about some specific values, warns about but ignores some other values, and/or rejects the use
+ * of some other values.
  *
- * @param <T> The type of the values of which certain are disallowed.
+ * @param <T> The type of the values of which certain are warned, ignored and/or disallowed.
  */
 public class Values<T> extends Guardrail
 {
+    private final Function<ClientState, Set<T>> warnedValues;
     private final Function<ClientState, Set<T>> ignoredValues;
     private final Function<ClientState, Set<T>> disallowedValues;
     private final String what;
 
     /**
      * Creates a new values guardrail.
      *
-     * @param ignoredValues    a {@link ClientState}-based of the values that are ignored.
-     * @param disallowedValues a {@link ClientState}-based of the values that are disallowed.
+     * @param warnedValues     a {@link ClientState}-based provider of the values for which a warning is triggered.
+     * @param ignoredValues    a {@link ClientState}-based provider of the values that are ignored.
+     * @param disallowedValues a {@link ClientState}-based provider of the values that are disallowed.
      * @param what             The feature that is guarded by this guardrail (for reporting in error messages).
      */
-    public Values(Function<ClientState, Set<T>> ignoredValues,
+    public Values(Function<ClientState, Set<T>> warnedValues,
+                  Function<ClientState, Set<T>> ignoredValues,
                   Function<ClientState, Set<T>> disallowedValues,
                   String what)
     {
+        this.warnedValues = warnedValues;
         this.ignoredValues = ignoredValues;
         this.disallowedValues = disallowedValues;
         this.what = what;
     }
 
     /**
-     * Triggers a warning for each of the provided values that are disallowed by this guardrail and triggers an action
-     * to ignore them. If the values are disallowed it will abort the operation.
+     * Triggers a warning for each of the provided values that is discouraged by this guardrail. If any of the values
+     * is disallowed it will abort the operation.
+     * <p>
+     * This assumes that there aren't any values to be ignored, thus it doesn't require an ignore action. If this is
+     * not the case and the provided value is set up to be ignored this will throw an assertion error.
+     *
+     * @param values The values to check.
+     * @param state  The client state, used to skip the check if the query is internal or is done by a superuser.
+     *               A {@code null} value means that the check should be done regardless of the query.
+     */
+    public void guard(Set<T> values, @Nullable ClientState state)
```

Comment:
nit: Might be useful to name this `guardWithoutIgnore()` or something similar for readability at the call location.